profit/main.py

```diff
@@ -46,34 +39,39 @@ def main():
     parser.add_argument('base_dir',
                         metavar='base-dir',
                         help='path to config file (default: current working directory)',
-                        default=getcwd(), nargs='?')
+                        default=default_base_dir, nargs='?')
     args = parser.parse_args()
 
     print(args)
 
-    """ Instantiate Config class from the given file """
-    config_file = safe_path_to_file(args.base_dir, default='profit.yaml')
-    config = Config.from_file(config_file)
+    """Instantiate Config from the given file."""
+    config_file = safe_path_to_file(args.base_dir, default=default_config_file)
+    config = BaseConfig.from_file(config_file)
 
     sys.path.append(config['base_dir'])
 
+    variables = VariableGroup(config['ntrain'])
+    vars = []
+    for k, v in config['variables'].items():
+        if isinstance(v, str):
+            vars.append(Variable.create_from_str(k, (config['ntrain'], 1), v))
+        else:
+            vars.append(Variable.create(**v))
+    variables.add(vars)
```

Comment:
That is a bit strange: While creating the `BaseConfig` the `Variable` objects were already built, but then replaced with a dictionary. Now the dictionary is converted into objects again.
I guess it's fine for now (as probably some other lines depend on the dictionary structure), but in the future we should consider completing the transformation into an object oriented config (see issue #143)